### PR TITLE
Use DOMAIN when set for certs, kube, and run:

### PR DESCRIPTION
**Pull request by HartS**
This contains the commit used from https://github.com/SUSE/scf/pull/1093

- Clean up case logic around kube vs. helm in make/kube
- Use domain resolution to set external_ip with make run
- Prefer preset DOMAIN for make/run
- Temporarily replace DOMAIN in dotenv file for make/kube and 'make
  certs'

As I point out in the SCF pull request, fissile still hardcodes 192.168.77.77 for external_ip, so that IP will still be used when making kube manifests

### CODE REVIEW

File: make/kube [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
-ENV_FILES="$(echo env/*.env)"
-
-CREATE_HELM_CHART=false
-BUILD_TARGET=kube
+DOMAIN=${DOMAIN:-cf-dev.io}
 
 if [ "${1:-}" = "helm" ]; then
-    CREATE_HELM_CHART=true
     BUILD_TARGET=helm
+    CREATE_HELM_CHART=true
```
Comment (andreas-kupries) [inline make/kube]: This variable does not look to be used anymore. Could be removed.

File: Makefile [hunk 2]
```diff
@@ -6,7 +6,9 @@ endif
 build: certs releases images
 
 certs:
+	[ -n ${DOMAIN} ] && sed -i "s/^DOMAIN=.*/DOMAIN=${DOMAIN}/g" ${GIT_ROOT}/env/defaults.env
```
Comment (andreas-kupries) [inline Makefile]: ""-quote DOMAIN.
Further, add `-e` to indicate the sed-script.
Also maybe move the `-i` to in front of the path, like in the other `sed`-invokations I saw.

File: make/kube [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
     ENV_FILES=env/certs.env
+else
+    BUILD_TARGET=kube
+    CREATE_HELM_CHART=false
```
Comment (andreas-kupries) [inline make/kube]: Ditto

File: make/kube [hunk 4]
```diff
@@ -1,20 +1,23 @@
...
+    ENV_FILES="$(echo env/*.env)"
+    if [ -n $DOMAIN ]; then
```
Comment (andreas-kupries) [inline make/kube]: ""-quote DOMAIN

File: make/kube [hunk 5]
```diff
@@ -1,20 +1,23 @@
...
 
 if [ "${1:-}" = "helm" ]; then
-    CREATE_HELM_CHART=true
     BUILD_TARGET=helm
+    CREATE_HELM_CHART=true
     ENV_FILES=env/certs.env
+else
+    BUILD_TARGET=kube
+    CREATE_HELM_CHART=false
+    ENV_FILES="$(echo env/*.env)"
```
Comment (andreas-kupries) [inline make/kube]: `${GIT_ROOT}` prefix missing.

File: make/kube [hunk 6]
```diff
@@ -1,20 +1,23 @@
...
+    if [ -n $DOMAIN ]; then
+      sed -e "s@^DOMAIN=.*@DOMAIN=$DOMAIN@" -i env/defaults.env
```
Comment (andreas-kupries) [inline make/kube]: `${GIT_ROOT}` prefix missing.

File: make/kube [hunk 7]
```diff
@@ -24,3 +27,4 @@ fissile build "${BUILD_TARGET}" \
     --output-dir="${BUILD_TARGET}" \
     --defaults-file="$(echo "${ENV_FILES}" | tr ' ' ,)" \
     --use-memory-limits=false
+git checkout env/defaults.env
```
Comment (andreas-kupries) [inline make/kube]: `${GIT_ROOT}` prefix missing.

File: make/run [hunk 8]
```diff
@@ -32,11 +32,13 @@ kubectl get storageclass persistent 2>/dev/null || {
     kubectl create -f -
 }
 
-source env/defaults.env
+DOMAIN=${DOMAIN:-$(grep '^DOMAIN=' env/defaults.env | sed 's/DOMAIN=//')}
```
Comment (andreas-kupries) [inline make/run]: `${GIT_ROOT}` prefix missing.

File: make/run [hunk 9]
```diff
@@ -32,11 +32,13 @@ kubectl get storageclass persistent 2>/dev/null || {
...
+source <(sed '/^DOMAIN=/d' env/defaults.env)
```
Comment (andreas-kupries) [inline make/run]: `${GIT_ROOT}` prefix missing.

File: make/run [hunk 10]
```diff
@@ -32,11 +32,13 @@ kubectl get storageclass persistent 2>/dev/null || {
...
 helm install helm \
      --namespace "${NAMESPACE}" \
      --set "env.DOMAIN=${DOMAIN}" \
-     --set "env.UAA_ADMIN_CLIENT_SECRET=${UAA_ADMIN_CLIENT_SECRET}"
+     --set "env.UAA_ADMIN_CLIENT_SECRET=${UAA_ADMIN_CLIENT_SECRET}" \
+     --set "kube.external_ip=$(getent hosts ${DOMAIN} | awk '{ print $1; exit }')"
```
Comment (andreas-kupries) [inline make/run]: How portable is the `getent` command ? Does OSX have it ? Will have to ask @jandubois here

Comment (HartS) [inline make/run]: Ah.. no, it's not on Mac... good catch. I'll see if I can get this in a more portable way with dig

File: make/kube (hunk 7 above)
Comment (andreas-kupries) [inline make/kube]: Jan's comment applies here too, i.e. temp file over undoing whatever local changes the user had.

File: make/run (hunk 8 above)
Comment (andreas-kupries) [inline make/run]: Proposal: Go with `cd ${GIT_ROOT}` at the beginning, and remove all other references

File: make/kube [hunk 11]
```diff
@@ -6,17 +6,22 @@ GIT_ROOT=${GIT_ROOT:-$(git rev-parse --show-toplevel)}
...
 
-ENV_FILES="$(echo env/*.env)"
-
-CREATE_HELM_CHART=false
-BUILD_TARGET=kube
+DOMAIN=${DOMAIN:-}
+if test -n "$DOMAIN"; then
+    # If DOMAIN is set, place DOMAINT at the end of defaults.env temporarily
+    echo "DOMAIN=$DOMAIN" >> env/defaults.env
+    trap "sed -i '$d' env/defaults.env" EXIT
```
Comment (andreas-kupries) [inline make/kube]: /home/vagrant/scf/src/uaa-fissile-release/make/kube: line 13: d: unbound variable

Yes. The outer `"`-quoting causes interpolation of `$d`, `d` does not exist.
Has to be `\`-escaped.

**Pull request merged**